### DISCUSSION TITLE
Warp Memory Error On HashGrid

### DISCUSSION THREAD

**Discussion by cadop** [NVIDIA/warp#155] [Q&A] [answered]
I have this demo script. I am not able to create a hashgrid over the size of ~47,000

If m,n = 215x215 (46,225) the grid works. If it is 220x215 (47,300) it runs into:

```
    runtime.core.hash_grid_update_host(self.id, radius, ctypes.cast(points.ptr, ctypes.c_void_p), len(points))
OSError: exception: access violation writing 0x0000000000000000
```

```
import numpy as np
import warp as wp
#### Initialize WARP #####
wp.init()
device='cpu'

m, n = 220, 215
nagents = m*n
hash_radius = 0.5
grid = wp.HashGrid(dim_x=nagents, dim_y=1, dim_z=nagents, device=device)
_pos = np.asarray([np.array([0,0,0], dtype=np.double) 
                                    for x in range(m) 
                                    for y in range(n)])
_pos_wp = wp.array(_pos, device=device, dtype=wp.vec3)
grid.build(points=_pos_wp, radius=hash_radius)
```

I thought it might have been gpu ram so I switched to cpu, same behavior. 

How can i make a larger hashgrid?

**Comment by shi-eric** [comment, marked as answer]
Hi @cadop, my initial look at this indicates that you're running into integer overflows in Warp due to the use of signed integers to store size variables. Note that `(220*215)**2 = 2237290000`, `(215*215)**2=2136750625`, and `2**31-1 = 2147483647`, which falls between the two values. I (or someone else) will update when we have a fix in.

**Reply by mmacklin** [reply]
I would add that you shouldn't create a grid with `dim_x = dim_z = nagents`, that will be much too large, like creating a texture with resolution of `47300 x 47300`. I would recommend setting `dim_x = m`, `dim_z=n`.

**Reply by cadop** [reply]
Yep, it was supposed to nagents total, not mxn, mxn
